models/Publication.js

```diff
@@ -188,6 +189,84 @@ class Publication extends BaseModel {
     return await Publication.query().patchAndFetchById(id, { deleted: date })
   }
 
+  static async update (newPubObj /*: any */) /*: Promise<any> */ {
+    // Create metadata
+    const metadata = {}
+    metadataProps.forEach(property => {
+      metadata[property] = newPubObj[property]
+    })
+
+    // Fetch the Publication that will be modified
+    let publication = await Publication.query().findById(urlToId(newPubObj.id))
+    if (!publication) {
+      return null
+    }
+
+    const modifications = _.pick(newPubObj, [
+      'id',
+      'name',
+      'description',
+      'datePublished',
+      'json',
+      'readingOrder',
+      'resources',
+      'links'
+    ])
+
+    if (metadata) {
+      modifications.metadata = metadata
+    }
+
+    if (modifications.readingOrder) {
+      modifcations.readingOrder = { data: modifications.readingOrder }
+    }
+    if (modifications.links) modifications.links = { data: modifications.links }
+    if (modifications.resources) {
+      modifications.resources = { data: modifications.resources }
+    }
+
+    // Assign the modifications to the publication object
+    publication = Object.assign(publication, modifications)
+
+    // Update Attributions if necessary
+    if (newPubObj.author) {
```

Comment:
I don't see where that newPubObj comes from. 
was the previous line supposed to be newPubObj = Object.assign.... ? If so, that should be moved later. You should update the attributions only if they are in the modifications.